### PR TITLE
Revoke & Renew implementation

### PR DESCRIPTION
**Pull request by bryanapellanes-okta**
- Properly implemented RevokeAsync
- Properly implemented RenewAsync
- Added RequestException event
- Added test for RequestException event
- Updated demo application to show sign-in, sign-out, revoke and renew examples.

See demo here: https://okta.circlehd.com/view/my-recording-jul-6-2021-7Erix32F3x

### CODE REVIEW

File: Okta.Xamarin/Okta.Xamarin/RequestExceptionEventArgs.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
+﻿using System;
```
Comment (laura-rodriguez) [inline Okta.Xamarin/Okta.Xamarin/RequestExceptionEventArgs.cs:1]: nit: header is missing.

File: Okta.Xamarin/Okta.Xamarin/ViewModels/RenewCommand.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
+﻿using System;
```
Comment (laura-rodriguez) [inline Okta.Xamarin/Okta.Xamarin/ViewModels/RenewCommand.cs:1]: Missing header.

File: Okta.Xamarin/Okta.Xamarin/OktaStateManager.cs [hunk 3]
```diff
@@ -153,20 +166,27 @@ public async Task<T> GetUserAsync<T>(string authorizationServerId = "default")
...
-            RenewResponse = renewResponse;
-            TokenType = renewResponse.TokenType;
-            AccessToken = renewResponse.AccessToken;
-            RefreshToken = renewResponse.RefreshToken;
-            if(refreshIdToken && !string.IsNullOrEmpty(renewResponse?.IdToken))
+            return await this.RenewAsync(this.RefreshToken, refreshIdToken, authorizationServerId);
+        }
+
+        /// <inheritdoc/>
+        public async Task<RenewResponse> RenewAsync(string refreshToken, bool refreshIdToken = false, string authorizationServerId = "default")
```
Comment (laura-rodriguez) [inline Okta.Xamarin/Okta.Xamarin/OktaStateManager.cs:179]: Do we have tests that cover this method?

File: Okta.Xamarin/Okta.Xamarin/OktaStateManager.cs [hunk 4]
```diff
@@ -175,11 +195,11 @@ public async Task<RenewResponse> RenewAsync(bool refreshIdToken = false, string
...
-            AccessToken = string.Empty;
-            IdToken = string.Empty;
-            RefreshToken = string.Empty;
-            Scope = string.Empty;
-            Expires = null;
+            this.AccessToken = string.Empty;
+            this.IdToken = string.Empty;
+            this.Scope = string.Empty;
+            this.Expires = null;
+            this.RefreshToken = string.Empty;
```
Comment (laura-rodriguez) [inline Okta.Xamarin/Okta.Xamarin/OktaStateManager.cs:202]: Do we have tests that cover this method?

**Review by laura-rodriguez — APPROVED** [review]
LGTM. I suggest adding more test coverage if we don't have tests that cover renew/revoke already.

**Comment by bryanapellanes-okta** [comment]
> LGTM. I suggest adding more test coverage if we don't have tests that cover renew/revoke already.

There should already be basic tests to ensure events are fired.  I'm planning on doing a test coverage pass at some point after the next release.

**Pull request merged**